### PR TITLE
Unit Tests ✍️

### PR DESCRIPTION
**Pull request by hsztan**
Our team of devs creates 🐣 unit tests for each resource 🥚, also known as a Class 🐓. The library used to accomplish this milestone is RSPEC. All test ⚙️ files were created in an individual folder 📂 called `spec` to maintain integrity and structure 🏗 for run-on testing. This will provide for later ⌛️ code refactoring and also make sure that the methods and code still work 🏋️‍♂️ as expected at a later date. 😇

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    person = Person.new('18', 'John', parent_permission: true)
+    expect(person.rentals).to eq([])
+  end
+
+  it 'should be able to rent a book' do
+    person = Person.new('18', 'John', parent_permission: true)
+    book = Book.new('The Great Gatsby', 'F. Scott Fitzgerald')
+    rental = Rental.new('2018-01-01', book, person)
+    expect(person.rentals).to eq([rental])
+  end
```
Comment (Gambit142) [inline spec/person_spec.rb:38]: - [x] Great work writing tests for the `Person` class. Kindly write tests for all methods in each class. In the `Person` class, that would be the `can_use_services` and `correct_name` methods. Kindly implement this. 👍

File: spec/teacher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  it 'should have an age' do
+    teacher = Teacher.new(24, 'Mathematician', 'John')
+    expect(teacher.age).to eq(24)
+  end
+
+  it 'should have a specialization' do
+    teacher = Teacher.new(24, 'Mathematician', 'John')
+    expect(teacher.specialization).to eq('Mathematician')
+  end
```
Comment (Gambit142) [inline spec/teacher_spec.rb:22]: - [x] Great work writing tests for the `Teacher` class. Kindly write tests for all methods in each class. In the `Teacher` class, that would be the `can_use_services` method. Kindly implement this. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @hsztan @Retky 👋, 

Great work team!

## Highlights
- Wrote tests using `Rspec` 💯 
- Wrote tests for the initialize method of most classes 💯 
- One test file per class 💯 
- Uses `describe` and `it` block correctly 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### I am writing this comment here because I can not highlight the necessary codes in the files changes section.

- Kindly write tests for the initialize method and all other method for the following classes:
  - Decorator
  - Nameable
  - CapitalizeDecorator
  - TrimmerDecorator

- _Kindly check the comments under the review for other required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by t-yanick — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**